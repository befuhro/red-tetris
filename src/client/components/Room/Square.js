import React from 'react';

const squareStyle = {
    height: '2.5vh',
    width: '2.5vh',
    borderStyle: 'outset',
    borderWidth: '0.2vh',
    borderColor: '#749099'
};

const realWhiteSquare = {backgroundColor: '#ffffff'};
const whiteSquare = {backgroundColor: '#5a6978'};
const redSquare = {backgroundColor: '#e12108'};
const darkblueSquare = {backgroundColor: '#262ee6'};
const yellowSquare = {backgroundColor: '#feef0c'};
const greenSquare = {backgroundColor: '#76f541'};
const lightblueSquare = {backgroundColor: '#16e3dc'};
const purpleSquare = {backgroundColor: '#b500ec'};
const orangeSquare = {backgroundColor: '#ff9b00'};

export default ({color, size}) => {
    switch (color) {
        case 'realwhite':
            return <div style={{...squareStyle, ...realWhiteSquare, ...size}} > </div>;
        case 'red':
            return <div style={{...squareStyle, ...redSquare, ...size}} > </div>;
        case 'darkblue':
            return <div style={{...squareStyle, ...darkblueSquare, ...size}} > </div>;
        case 'yellow':
            return <div style={{...squareStyle, ...yellowSquare, ...size}} > </div>;
        case 'green':
            return <div style={{...squareStyle, ...greenSquare, ...size}} > </div>;
        case 'lightblue':
            return <div style={{...squareStyle, ...lightblueSquare, ...size}} > </div>;
        case 'purple':
            return <div style={{...squareStyle, ...purpleSquare, ...size}} > </div>;
        case 'orange':
            return <div style={{...squareStyle, ...orangeSquare, ...size}} > </div>;
        default:
            return <div style={{...squareStyle, ...whiteSquare, ...size}} > </div>;
    }
}
