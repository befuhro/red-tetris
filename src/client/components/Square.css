.square
{
    height: 25px;
    width: 25px;
    background-color: white;
    margin: 1px;
}

.red
{
    background-color: darkred;
}