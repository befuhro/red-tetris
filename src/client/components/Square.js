import React from 'react';
import './Square.css';


function changeColor(id) {
    console.log('ID = ', id);
}

function renderSquare(id, color) {
    if (color !== undefined && color === 'red') {
        return (<div id={id} className="square red" onClick={() => changeColor(id)}></div>);
    }
    return (<div id={id} className="square" onClick={() => changeColor(id)}></div>);
}

const Square = ({id, color}) => (
    renderSquare(id, color)
);

export default Square;