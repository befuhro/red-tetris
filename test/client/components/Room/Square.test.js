import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({adapter: new Adapter()});

import Square from '../../../../src/client/components/Room/Square';

describe('<Square/>', () => {
    describe('white square', () => {
        const component = enzyme.mount(
            <Square
                color={'realwhite'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('red square', () => {
        const component = enzyme.mount(
            <Square
                color={'red'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('darkblue square', () => {
        const component = enzyme.mount(
            <Square
                color={'darkblue'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('yellow square', () => {
        const component = enzyme.mount(
            <Square
                color={'yellow'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('green square', () => {
        const component = enzyme.mount(
            <Square
                color={'green'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('lightblue square', () => {
        const component = enzyme.mount(
            <Square
                color={'lightblue'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('purple square', () => {
        const component = enzyme.mount(
            <Square
                color={'purple'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('orange square', () => {
        const component = enzyme.mount(
            <Square
                color={'orange'}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });

    describe('default square', () => {
        const component = enzyme.mount(
            <Square
                color={''}
                size={''}
            />);
        it('should render white square', () => expect(component.html()).toMatchSnapshot());
    });
});
