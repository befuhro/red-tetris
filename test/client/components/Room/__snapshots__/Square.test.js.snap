// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square/> darkblue square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(38, 46, 230);\\"> </div>"`;

exports[`<Square/> default square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div>"`;

exports[`<Square/> green square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(118, 245, 65);\\"> </div>"`;

exports[`<Square/> lightblue square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div>"`;

exports[`<Square/> orange square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(255, 155, 0);\\"> </div>"`;

exports[`<Square/> purple square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(181, 0, 236);\\"> </div>"`;

exports[`<Square/> red square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(225, 33, 8);\\"> </div>"`;

exports[`<Square/> white square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(255, 255, 255);\\"> </div>"`;

exports[`<Square/> yellow square should render white square 1`] = `"<div style=\\"height: 2.5vh; width: 2.5vh; border-style: outset; border-width: 0.2vh; border-color: #749099; background-color: rgb(254, 239, 12);\\"> </div>"`;
